### ISSUE TITLE
Prediction interrupted; please retry (code: PA)

### ISSUE THREAD

**Issue by Soykertje** [replicate/replicate-python#135]
When I try to consume one of my private endpoints like this:
`result = replicate.run("my-user/model-name:version",
                               input={"file": open(file_path, "rb")})`

I'm getting that exception but when I execute it from the [website](https://replicate.com) it works with no problem.

**Comment by mattt** [comment]
Hi @Soykertje. Can you please share a link to a prediction that failed and one that succeeded? Or else, share the full error message / logs, or anything else that we could use to debug your problem?

**Comment by awerks** [comment]
I aslo got this problem upon executing via API. The website works normally. This is the only message it shows, there are no other logs. It happens with large files > 20 MB
 File "C:\Users\susti\AppData\Local\Programs\Python\Python311\Lib\site-packages\replicate\client.py", line 140, in run
    raise ModelError(prediction.error)
replicate.exceptions.ModelError: Prediction interrupted; please retry (code: PA)

**Comment by mattt** [comment]
@awerks @Soykertje On the website, files are uploaded to a URL which is passed as an input to the prediction. When you set a file as an input through a client library, the file is sent in the request body as a base64-encoded `data:` URI. Replicate's API enforces limits on the size of requests, which may be causing this error.

**Comment by awerks** [comment]
@mattt is there a workaround to this?

**Comment by Soykertje** [comment]
Hello @mattt since they are predictions made to a private endpoint, apparently you'll be unable to access them, in the case you're able to access them, here they are: [worked](https://replicate.com/p/eyrrhwrbcrhmpkkn75ubzfwwdy) and [failed](https://replicate.com/p/qi7vqmbbn6ly2cb4xchwg5hnea)

Those are my [predictions](https://imgur.com/a/2V70Ja0). As you can notice, it's a custom implementation of whisper (since the implemented endpoint by Replicate does not include some kwargs that I need and uses an old version). The prediction that worked looks like [this](https://imgur.com/a/US4aQcb) and when a prediction failed related to this problem, looks like [this](https://imgur.com/a/NMZOu2M).

**Comment by mattt** [comment]
@awerks Yes, you can upload the file to a URL and pass that URL as the input for your prediction.

@Soykertje Thanks for sharing those. I'll take a look.

**Comment by awerks** [comment]
@Soykertje, it works via a URL, but it the downloading time is obviously billed

**Comment by Soykertje** [comment]
@awerks Do you have any idea of how slow? I'm planning to use files that in the 95 % (or above) of the cases will not be bigger than 200 MB, so depending on the download speed, some billed seconds are still worth.

**Comment by awerks** [comment]
@Soykertje It is not slow, but it adds up. 5-7 seconds each time could be a deal-breaker for someone

**Comment by Soykertje** [comment]
@awerks Uhhhh I made some math and yeah, absolutely... Depending on the volume of requests, the cost of those seconds can end up being something to pay attention.

@mattt Is there any reason for having such a low payload size configuration into the API's web server and not in the website?

**Comment by brassel** [comment]
Hi replicate team, thanks for this great platform!

I have stumbled across the same issue and for me reducing the size of the input files was very easy.
After that it worked just fine. 
But a more informative error message would really be appreciated, also by others I am sure.

You could also point me to the code where this is defined and I could make improving the error my first contribution, maybe?

Thank you and kind regards

**Comment by neal-qualitative** [comment]
@mattt Passing in a url as input instead of a base64 encoded file is working for me. However, the files that my platform is processing cannot be accessible through a public url. How can we identify replicate in order to restrict access to model input files?

For example, if I'm storing model input files in s3, is there a replicate s3 user that I can grant access to?

**Comment by brassel** [comment]
@neal-qualitative, in GCloud, I can create a "signed URL" for short term/single use access purposes like yours. Maybe something like that is possible in AWS and would be a solution for you?

**Comment by ryx2** [comment]
I just got this issue, and all of the inputs were strings. the code does involve downloading some files from s3, but those files are collectively about 40 mb, and the compute time is about 70s before failing with the exact error mentioned above. somehow about 1/2 of the jobs (about 20) i created in the last 5 minutes ran into this exact error. this is not something I've seen before right now, even for this same model and version.

**Comment by ancri** [comment]
Is there an update on what kind of image size is expected to cause an issue, and what size we should expect is safe to pass as base64 encoded?

**Comment by sachsom95** [comment]
I deploy it as an S3 presignedURL and it solves the issue for me

**Comment by Soykertje** [comment]
@ryx2 Have you tried running the prediction from the web? The web uses the API too so what it does is upload the files into an internal CDN and sends the files as download links to the model... In other words, it's basically no difference between sending links using the API or running it from the web, if the prediction fails from the web too, consider opening a new issue because your problem would not be related to sending files in the body.

**Comment by mazperez** [comment]
me aparaece ete mensage y no se como puedo solucionarlo.Prediction failed.

cannot identify image file '/tmp/tmp3j8zxxmsguillermofotos.zip'

Generated in
